### PR TITLE
Bug 1839640 - Add new metric type: object

### PR DESCRIPTION
**Pull request by badboy**
This is the most bare-bones version for now.
It validates the `structure` and then generates the corresponding Rust code.

### CODE REVIEW

File: tests/data/object.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            properties:
+              count:
+                type: number
+              string:
+                type: string
+
+crash.stack:
+  threads:
+    type: object
+    description: All threads' frame information.
```
Comment (chutten) [inline tests/data/object.yaml:73]:
```suggestion
    description: |
      All threads' frame information.
      e.g. `[{frames: [{module_index: 0, ip: "0xdecafc0ffe", trust: "scan"}]}]`
```

**Review by chutten — CHANGES_REQUESTED** [review]
Is the test that the yaml is parsed at all? Is there a test of the rust codegen?

**Review by travis79 — APPROVED** [review]
Looks good to me, I guess we will add support for other languages as needed?

**Comment by badboy** [comment]
> Looks good to me, I guess we will add support for other languages as needed?

yes. I'll file bugs.

**Pull request merged**